cmd/root.go

```diff
@@ -52,83 +48,87 @@ func init() {
 	rootCmd.PersistentFlags().StringVarP(&cfgFile, "config", "c", "", "config file (default is $HOME/.weep.yaml)")
 	rootCmd.PersistentFlags().StringSliceVarP(&assumeRole, "assume-role", "A", make([]string, 0), "one or more roles to assume after retrieving credentials")
 	rootCmd.PersistentFlags().StringVar(&logFormat, "log-format", "", "log format (json or tty)")
+	rootCmd.PersistentFlags().StringVar(&logFile, "log-file", viper.GetString("log_file"), "log file path")
 	rootCmd.PersistentFlags().StringVar(&logLevel, "log-level", "", "log level (debug, info, warn)")
-	rootCmd.PersistentFlags().BoolVarP(&runAsService, "svc", "s", false, "run weep as a service")
 }
 
 func Execute() {
-	shutdown := make(chan os.Signal, 1)
+	shutdown = make(chan os.Signal, 1)
 	done = make(chan int, 1)
-	signal.Notify(shutdown, syscall.SIGINT, syscall.SIGTERM)
-	if runAsService {
-		RunService()
-	} else {
-		if err := rootCmd.Execute(); err != nil {
-			_, _ = fmt.Fprintln(os.Stderr, err)
-			os.Exit(1)
-		}
+	signal.Notify(shutdown, syscall.SIGINT, syscall.SIGTERM, os.Interrupt)
+
+	if err := rootCmd.Execute(); err != nil {
+		rootCmd.PrintErr(err)
 	}
 }
 
+// initConfig reads in configs by precedence, with later configs overriding earlier:
+//   - embedded
+//   - /etc/weep/weep.yaml
+//   - ~/.config/weep/weep.yaml
+//   - ~/.weep.yaml
+//   - ./weep.yaml
+//   - CLI arg-specified file
 func initConfig() {
-	if cfgFile != "" {
-		viper.SetConfigFile(cfgFile)
-	} else {
-		home, err := homedir.Dir()
-		if err != nil {
-			fmt.Println(err)
-			os.Exit(1)
-		}
-		viper.SetConfigType("yaml")
-		viper.SetConfigName(".weep")
-		viper.AddConfigPath(".")
-		viper.AddConfigPath(home)
-		viper.AddConfigPath(home + "/.config/weep/")
+	home, err := homedir.Dir()
+	if err != nil {
+		log.Fatal(err)
 	}
+	viper.SetConfigType("yaml")
 
+	// Read embedded config if available
 	if err := config.ReadEmbeddedConfig(); err != nil {
 		log.Debugf("unable to read embedded config: %v; falling back to config file", err)
 	}
 
-	if err := viper.ReadInConfig(); err != nil {
-		if _, ok := err.(viper.ConfigFileNotFoundError); ok && config.EmbeddedConfigFile != "" {
-			log.Debugf("no config file found, trying to use embedded config")
-		} else if isatty.IsTerminal(os.Stdout.Fd()) {
-			err = util.FirstRunPrompt()
-			if err != nil {
-				log.Fatalf("config bootstrap failed: %v", err)
-			}
-		} else {
-			log.Debugf("unable to read config file: %v", err)
+	// Read in config from etc
+	viper.SetConfigName("weep")
+	viper.AddConfigPath("/etc/weep/")
+	_ = viper.MergeInConfig()
+
+	// Read in config from config dir
```

Comment:
We need to determine if any debugging information is printed to the console when this happens, in case someone is unaware that they have multiple configuration files being merged. If it is not printed, I'd want to log it here.

Reply:
It's not, which I'm having a hard time trying to iron out. We have to do this before we configure logging since `initLogging` depends on config values for log level and file. Any ideas?

Reply:
I don't have a good solution for your comment (yet), but I did change the way the config merging is handled a bit. Now a CLI-specified config file will be the _only_ configuration used (even if there's an embedded config).